presto-hive/src/main/java/com/facebook/presto/hive/HudiDirectoryLister.java

```diff
@@ -132,7 +133,8 @@ public boolean hasNext()
         public HiveFileInfo next()
                 throws IOException
         {
-            FileStatus fileStatus = hoodieBaseFileIterator.next().getFileStatus();
+            HoodieBaseFile baseFile = hoodieBaseFileIterator.next();
+            FileStatus fileStatus = baseFile.getBootstrapBaseFile().map(BaseFile::getFileStatus).orElse(baseFile.getFileStatus());
```

Comment:
Orthogonal question: can you please audit the code path that maps out the corresponding bootstrap base file for a given base file and see if it reads the bootstrap index efficiently? i.e do we read it once and reuse or open/close per base file (which could hurt perf?)